### DISCUSSION TITLE
Out of memory error in processing large images of size 50 MB

### DISCUSSION THREAD

**Discussion by rajatg64** [dlemstra/Magick.NET#579] [General] [answered]
### System Configuration
Code is running on AWS ECS container  CPU 512MB , Hard/Soft Memory Limits 1024 MiB

- Magick.NET version: Magick.NET-Q16-AnyCPU 7.15.1

### Question
I'm using MagickNet to process images and convert them to thumbnail. It works perfectly for images till 45 MB, But when i upload image of size 50 MB the aws ECS instance runs out of memory and the container restarts itself.

Please find the below code

  using (var image = new MagickImage(bytes))             <<-- This line throws out of memory error in case of large files 50 Mb
                    {
                        image.Resize(128, 128);
                        image.Strip();
                        image.Quality = 10;
}

**Comment by Jcomper** [comment]
Have you tried to set memory limit usage? [Documentation](https://github.com/dlemstra/Magick.NET/blob/master/docs/Readme.md#default-memory-limit) 

Like this:
`ImageMagick.ResourceLimits.LimitMemory(new Percentage(10));`

**Comment by dlemstra** [comment]
Can we close this issue @rajatg64?

**Comment by manjunthcm** [comment]
hi all
im using Magick.Net-Q16-AnyCPU ( v7.16.0)
i am able to load tiff images upto 368MB without any Settings.
The image load was failing after this Size. So i tried setting memory limit usage, as ResourceLimits.LimitMemory(new Percentage(80));
even then it is not working.

any help is highly appreciated.

**Comment by dlemstra** [comment]
The size of the file does not really matter. The number of pixels determines how much memory you need. Can you clarify what you mean by "not working"? And can you share a file that demonstrates this issue.

**Comment by manjunthcm** [comment]
Thanks for the quick reply.
My image is of 16000x10000 pixels size image. When i try to load this with the following code, 


ResourceLimits.LimitMemory(new Percentage(100));
MagickImage userTiffImage = new MagickImage(SelectedTiffImageName) ;

it throws me an exception, 
"memory allocation failed `..\\Original_CMYK_imgs\\Sample8.tif' @ error/tiff.c/ReadTIFFImage/1836"

**Comment by manjunthcm** [comment]
just an update, the Same file im able to load using ImageMagick command line.

**Comment by dlemstra** [comment]
In what kind of environment are you running this? Your images probably requires `16000*10000*32*3 bits = 1.92Gb`  of memory. This will not work if you run the application in a 32bits environment.

**Comment by manjunthcm** [comment]
Thanks for the reply.
My System configuration is : Windows 10x64 , Version 1809, i7 2.6GHz, 16GB RAM 500GB HDD with around 100GB free Space

The image im trying to load is a CMYK image, so i understand the Size to be 16000x10000x4. 
Also, i can load same image and resize it to smaller size using command line option.
I am facing only problem with NuGet package.

**Comment by dlemstra** [comment]
If you have a CMYK image you will indeed need to multiply it by 4. And that makes the total required memory 2.56 GB Is your .NET application running in a 32bit environment? That is the only reason I could think of why this would not work in your .NET application. What happens when you switch to the `Magick.NET-Q16-x64` version of this library?

**Comment by manjunthcm** [comment, marked as answer]
yes, my .Net application is in AnyCPU. 
So now instead of Magick.Net-Q16-AnyCPU, i used Magick.Net-Q16-x64 , and now im able to load the image.

**Comment by manjunthcm** [comment]
Magick.Net-Q16-AnyCPU is supposed to support both x64 and x86, 
then why i have issue [ memory issue only with bigger images ] only with Magick.Net-Q16-AnyCPU and not with Magick.Net-Q16-x64

**Reply by dlemstra** [reply]
This is happening because you are running your application in 32 bit mode. When you run an application in AnyCPU the application will decide to either run in 32 or 64 bits. And when the application is running in 32 bit mode you will have memory limitations: https://docs.microsoft.com/en-us/windows/win32/memory/memory-limits-for-windows-releases.

**Comment by manjunthcm** [comment]
many thanks..
